Algorithms.Tests/Numeric/EulerMethodTest.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using Algorithms.Numeric;
+using NUnit.Framework;
+using System.Collections.Generic; 
+
+namespace Algorithms.Tests.Numeric
+{
+    public static class EulerMethodTest
+    {
+        [Test]
+        public static void TestLinearEquation()
+        {
+            Func<double, double, double> exampleEquation1 = (x, y) => x;
+            List<double[]> points =  Algorithms.Numeric.EulerMethod.EulerFull(0, 4, 0.1, 0, exampleEquation1);
+            double yEnd = points[points.Count - 1][1];
+            Assert.AreEqual(yEnd, 7.800000000000003);

```

Comment:
Why is it 7.8? And use yEnd.Should().BeApproximately(7.8).
y' = x
y = x^2 / 2 + C
Taking into account that y(0) = 0, C = 0
y = x^2 / 2
Then y(4) = 4^2 / 2 = 8, not 7.8

Reply:
It should be 8 but that's the error due to the algorithm itself. I have decreased the step-size to get it closer.